cwltool/workflow_job.py

```diff
@@ -559,15 +564,31 @@ def receive_output(
         step: WorkflowJobStep,
         outputparms: List[CWLObjectType],
         final_output_callback: OutputCallbackType,
+        runtimeContext: RuntimeContext,
+        container_engine: str,
         jobout: CWLObjectType,
         processStatus: str,
     ) -> None:
 
+        requirements = {
+            **{h["class"]: h for h in step.tool.get("hints", [])},
+            **{r["class"]: r for r in step.tool.get("requirements", [])},
+        }
```

Comment:
Is there a reason to do this instead of calling `step.tool.get_requirement("http://commonwl.org/cwltool#Loop")` ?

Reply:
No, simply didn't know the existence of that method XD. Fixed by calling `step.step.get_requirement`.